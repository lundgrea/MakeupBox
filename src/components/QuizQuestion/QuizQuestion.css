.quiz-question-answers {
  margin: 10px 10px 10px 10px;
  text-align:left;
  display:inline;
}

.quiz-question{
  font-size: 20px;
}

.quiz-question-full {
  margin: 40px 10px 40px 10px;
  background-color: white;
  padding: 10px 10px 20px 10px;
  border-radius: 2px;
}