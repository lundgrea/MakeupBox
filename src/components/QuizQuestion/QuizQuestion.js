import React from 'react';
import './QuizQuestion.css';
import PropTypes from 'prop-types';


export const QuizQuestion = ({answers, question, id, name, handleChange}) => {
  const buildAnswers = () => {
    let answersList = answers.map((answer, index) => {
    return (
      <div className='quiz-question-answers' key={id + `${index}`}>
        <input
          type='radio'
          name={name}
          value={answer}
          key={id}
          id={id + `${index}`}
          onChange={handleChange}
        ></input>
        <label htmlFor={id + `${index}`}>{answer}</label>
      </div>
    );
    });
    return answersList
  };

  return (
    <article className='quiz-question-full'>
      <h3 className='quiz-question'>{question}</h3>
      {buildAnswers()}
    </article>
  );
};

export default QuizQuestion;

QuizQuestion.propTypes = {
  answers:PropTypes.array.isRequired,
  handleChange:PropTypes.func.isRequired,
  id:PropTypes.number.isRequired,
  name:PropTypes.string.isRequired,
  question:PropTypes.string.isRequired,
};