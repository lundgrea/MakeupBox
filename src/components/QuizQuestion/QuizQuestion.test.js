import React from 'react';
import { shallow } from 'enzyme';
import QuizQuestion from './QuizQuestion';

describe('QuizQuestion', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<QuizQuestion 
     question='What is your primary interest?'
     answers={['lips', 'eyes']}
     key={1}
     name='interests'
     handleChange={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot();
  });
});