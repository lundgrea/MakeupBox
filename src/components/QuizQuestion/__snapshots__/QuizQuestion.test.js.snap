// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizQuestion should match the snapshot with all data passed in correctly 1`] = `
<article
  className="quiz-question-full"
>
  <h3
    className="quiz-question"
  >
    What is your primary interest?
  </h3>
  <div
    className="quiz-question-answers"
    key="undefined0"
  >
    <input
      name="interests"
      onChange={[MockFunction]}
      type="radio"
      value="lips"
    />
    <label
      htmlFor="lips"
    >
      lips
    </label>
  </div>
  <div
    className="quiz-question-answers"
    key="undefined1"
  >
    <input
      name="interests"
      onChange={[MockFunction]}
      type="radio"
      value="eyes"
    />
    <label
      htmlFor="eyes"
    >
      eyes
    </label>
  </div>
</article>
`;
